src/main/io/osd.c

```diff
@@ -687,6 +688,18 @@ static bool osdDrawSingleElement(uint8_t item)
             return true;
         }
 
+    case OSD_G_FORCE:
+        {
+            float osdGForce = 0;
+            for (int axis = 0; axis < 3; axis++) {
+                const float a = accAverage[axis];
+                osdGForce += a * a;
+            }
+            osdGForce = sqrtf(osdGForce) / acc.dev.acc_1G * 10;
```

Comment:
`sqrtf` is too expensive, we can't afford it here.
Use `pow_approx(osdGForce, 0.5)` instead.
Division is also unlikely to be optimized out as it requires static storage, so we should save reciprocal somewhere in `acc.dev.acc_1G_rcp` and use it here.

Reply:
Use `pow_approx(osdGForce, 0.5)` instead.
Division is also unlikely to be optimized out as it requires static storage, so we should save reciprocal somewhere in `acc.dev.acc_1G_rcp` and use it here.

Reply:
Should I do the reciprocal in this PR? I can see a lot of other places that would benefit from it too, qualifying it as another PR.

Reply:
I can do it on my own if you like, just get rid of `sqrtf` :)

Reply:
done :)

Reply:
there is 14-tick sqrt operation on FPU (M4/M7), so it is not cheap, but considerably cheaper than pow_approx.
pow_approx was implemented for baro altitude calculations. It isn't speed optimized, it was introduced to avoid libm `pow` implementation (which pulls in double precision math library)

Reply:
@ledvinap could you help/show me how to call this FPU sqrt operation? maybe hit me up on slack @leocb

Reply:
`sqrt` gets translated into this instruction.
I would prefer displaying `G_squared` in OSD rather than G to avoid `sqrt` altogether

Reply:
`sqrt ` overflows the available F3 flash, hence why I used `sqrtf`.

Displaying g squared would not be intuitive enough and would diverge to much from the standard G representation people are used to.

I'm looking at faster ways to to this kinda of math, since precision is not important here, maybe we can use a sqrt reciprocal.. I'll keep looking

Reply:
`sqrtf` presumably uses the same Cortex instruction as well

Reply:
@DieHertz : Avoiding sqrtf is over-optimization. Just get correct result.
OSD should not be on critical path anyway ...